airgap/v2/cmd/client/download/download.go

```diff
@@ -16,100 +16,94 @@ package download
 
 import (
 	"context"
+	"encoding/csv"
 	"fmt"
 	"io"
 	"os"
 	"strings"
 
 	"github.com/go-logr/logr"
-	"github.com/go-logr/zapr"
 	"github.com/redhat-marketplace/redhat-marketplace-operator/airgap/v2/apis/fileretreiver"
 	v1 "github.com/redhat-marketplace/redhat-marketplace-operator/airgap/v2/apis/model/v1"
+	"github.com/redhat-marketplace/redhat-marketplace-operator/airgap/v2/cmd/client/util"
 	"github.com/spf13/cobra"
-	"github.com/spf13/viper"
-	"go.uber.org/zap"
 	"google.golang.org/grpc"
-	"google.golang.org/grpc/credentials"
 )
 
 type DownloadConfig struct {
-	fileName        string
-	fileId          string
-	outputDirectory string
-	conn            *grpc.ClientConn
-	client          fileretreiver.FileRetreiverClient
+	fileName         string
+	fileId           string
+	outputDirectory  string
+	fileListPath     string
+	deleteOnDownload bool
+	conn             *grpc.ClientConn
+	client           fileretreiver.FileRetreiverClient
+	log              logr.Logger
 }
 
-var (
-	dc  DownloadConfig
-	log logr.Logger
-)
+var dc DownloadConfig
```

Comment:
With the use of a provider, just move this declaration to the DownloadCmd run func. For each command you'll just create a new downloadconfig struct using the provider and call the functions. No need for it to be global.